WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -19,7 +20,42 @@ def new
   end
 
   def index
-    @competitions = Competition.all.select([:id, :name, :cityName, :countryId]).order(:year, :month, :day).reverse_order
+    @regions = { 'Continent' => Continent.all.map { |continent| [continent.name, continent.id] },
+                 'Country' => Country.all.map { |country| [country.name, country.id] } }
+    @events = [ ["All", "all"], ["",""] ] + Event.all_official.map { |event| [event.name, event.id] }
+    @years = [ ["Current","current"],["All","all"],["",""] ] + Competition.where(showAtAll: true).map(&:year).uniq.sort.reverse!
+    @competitions = Competition.where(showAtAll: true).order(:year, :month, :day).reverse_order
+
+    # This need to be the first thing, otherwise @competitions will be an array instead of an object
+    # and the .where will not work
+    if params[:years].blank?
+      params[:years] = "current"
+    end
+
+    if params[:years] == "current"
+      @competitions = @competitions.where("CAST(CONCAT(year,'-',month,'-',day) as Datetime) > ?", (Date.today - PAST_COMPETITIONS_DAYS))
+    elsif params[:years] != "all"
+      @competitions = @competitions.reject { |competition| competition.year.to_s != params[:years] }
+    end
+
+    if params[:event] && params[:event] != "all"
+      @competitions = @competitions.reject { |competition| !competition.has_event?(Event.find(params[:event])) }
+    end
+
+    if params[:region] && params[:region] != "all"
```

Comment:
It would be more consistent with the way you checked `years` if you added something like this up by your guard for `:years`:

```ruby
if params[:event].blank?
  params[:event] = "all"
end
if params[:region].blank?
  params[:region] = "all"
end
```